### DISCUSSION TITLE
[BE101] 留言板 註冊功能重複帳號提示問題討論

### DISCUSSION THREAD

**Discussion by PT-HUANG** [Lidemy/forum#309] [後端相關問題] [answered]
老師及各位同學大家好
不曉得大家在上這堂課時有沒有遇到這樣的問題

輸入以下的程式碼：
```php
$result = $conn->query($sql);
  if (!$result) {
    $code = $conn->errno;
    if ($code === 1062) {
      header('Location: register.php?errCode=2');
    }
    die($conn->error);
  }

  header("Location: index.php");
``` 
假如輸入一樣的username，點選提交會跳出錯誤訊息：
![ERROR](https://github.com/Lidemy/forum/assets/125849885/92d9cfbc-247e-43f3-b943-f2f38a70e4ec)

猜想可能是在這行$conn->query($sql)
如果發現有錯誤就會自動導向$conn->connect_error

嘗試了許多解決方法，最後找出來的答案如下：

```php
  try {
    $result = $conn->query($sql);
  } catch (Throwable $e) {
    $code = $e->getCode();
    if($code === 1062) {
      header("Location:./register.php?err_code=2");
    }
    if(!$result) {
      die($conn->error);  
    }
  }

  header("Location:./index.php");
``` 

如果用print_r($e)會顯示一串的文字
再利用$e->getCode()，將mysqli_sql_expection這個物件中的code取值，就可以成功得到代碼了
![ERROR2](https://github.com/Lidemy/forum/assets/125849885/320e15e0-537d-4c43-85b6-f35f69012a1b)

不曉得是不是因為版本的關係，導致原來的$conn->errno沒辦法使用
如果有更好的解法或是意見，歡迎大家不吝分享交流
Google真的是很方便的東西XD

**Comment by aszx87410** [comment, marked as answer]
看起來應該是版本有差，導致錯誤處理的方式不太一樣了
